app/home/routes.py

```diff
@@ -120,14 +132,38 @@ def user_listing(user_id):
 @blueprint.route("/property/update/<int:property_id>", methods=["GET", "POST"])
 @login_required
 def update_property(property_id):
-    from app.tasks import update_prop_images
+    from app.tasks import delete_img_obj, image_process
+
+    prop_to_update = Property.query.get(property_id)
 
     form = UpdatePropertyForm()
     if request.method == "POST" and form.validate_on_submit():
         if request.files["prop_photos"].filename:
-            img_files = request.files.getlist("prop_photos")
-            temp_dir, image_file_list = save_images_to_temp_folder(img_files)
-            update_prop_images(temp_dir, image_file_list, property_id)
+            if request.files["prop_photos"].filename:
+                # delete previous images before
+                delete_img_obj.delay(
+                    bucket,
+                    s3_prop_img_dir + prop_to_update.image_folder,
+                    json.loads(prop_to_update.photos),
+                )
+
+                img_files = request.files.getlist("prop_photos")
+                folder_name = save_to_redis(img_files, current_user.username)
+
+                image_process.delay(folder_name, s3_prop_img_dir)
+                image_names = redis_client.hgetall(
+                    folder_name
+                )  # Returns dictionary object
+                image_list = [
+                    f"{image_name.decode('utf-8')}.jpg"
```

Comment:
same here I think after decoding the object it should carry its entire original name unless otherwise
which though we shouldnt limit the extension in case there are `.png` images that were uploaded etc

Reply:
I want the file name to be a unique string of characters because the image uploaded can contain unwanted characters.